docs/design_proposals/configurable-kubecontext.md

```diff
@@ -0,0 +1,183 @@
+# Configurable kubecontext
+
+* Author(s): Cornelius Weig (@corneliusweig)
+* Design Shepherd: Balint Pato (@balopat)
+* Date: 29 June 2019
+* Status: [Reviewed/Cancelled/Under implementation/Complete]
+
+## Background
+
+So far, Skaffold always uses the currently active kubecontext when interacting with a kubernetes cluster.
+This is problematic when users want to deploy multiple projects with different kubecontexts, because the user needs to manually switch the context before starting Skaffold.
+In particular when working on multiple such projects in parallel, the current behavior is limiting.
+
+Open issues concerning this problem are
+
+- Allow option to specify the kubectl context (#511)
+- Support kube.config and kube.context for specifying alternative Kubernetes config file or context (#2325)
+- Feature: Support regex in profile activation via kubeContext (#1677)
+- Skaffold.yaml files are not portable (#480)
+- Support forcing a context and a namespace for a profile/command (#2426)
+
+There also was an attempt to add a configuration option to `skaffold.yaml` (Support for overriding kubectl context during deployment #1540).
+
+The goal of this document is to create an agreement on what options should be supported and identify edge cases.
+
+## Design
+
+There are four places where kubecontext activation can be added:
+<table>
+    <thead>
+        <th>Precedence</th> <th>Kind</th> <th>Details</th>
+    </thead>
+    <tbody>
+        <tr>
+            <td>1. (highest)</td>
+            <td>CLI option</td>
+            <td>
+              The kubernetes standard to set the kubecontext is <code>--context</code>.
+              However, in Skaffold this term is so overloaded that it should more precisely be named <code>--kube-context</code>.
+              This flag is necessary for IDE integration.
+            </td>
+        </tr>
+        <tr>
+            <td>2.</td>
+            <td>Env variable</td>
+            <td>
+              <code>SKAFFOLD_KUBE_CONTEXT</code>, similar to other Skaffold flags.
+            </td>
+        </tr>
+        <tr>
+            <td>3.</td>
+            <td><code>skaffold.yaml</code></td>
+            <td>
+              Json-path <code>deploy.kubeContext</code>.
```

Comment:
```suggestion
              Json-path <code>[profile.]deploy.kubeContext</code>.
```

Reply:
My current feeling is that we should have the option also in the main pipeline. Although this may not be the recommended way to use this feature, I have a hunch that this will be requested nevertheless.

Reply:
agreed, `[profile.]` meant to signify main pipeline or optionally in a profile...maybe there is a better way to express that :)